### ISSUE TITLE
ROCM packages

### ISSUE THREAD

**Issue by isuruf** [conda-forge/staged-recipes#10123]
- ROCm Core Components
   - [ ] ROCk Kernel Driver: rock-dkms
   - [x] ROCr Runtime: rocr-runtime
   - [x] ROCt Thunk Interface: roct-thunk-interface

- ROCm Support Software
   - [ ] ROCm SMI: rocm-smi
   - [ ] ROCm cmake: rocm-cmake
   - [x] rocminfo: rocminfo
   - [x] ROCm Bandwidth Test: rocm-bandwidth-test

- ROCm Development Tools
   - [x] HCC compiler: hcc
   - [x] HIP: hip
   - [x] ROCm Device Libraries: rocm-device-libs
   - [x] ROCm OpenCL: rocm-opencl-runtime
   - [ ] ROCM Clang-OCL Kernel Compiler: rocm-clang-ocl
   - [ ] Asynchronous Task and Memory Interface (ATMI): atmi
   - [ ] ROCr Debug Agent: rocr_debug_agent
   - [x] ROCm Code Object Manager: rocm-comgr
   - [ ] ROC Profiler: rocprofiler-dev
   - [ ] ROC Tracer: roctracer-dev
   - [ ] Radeon Compute Profiler: rocm-profiler

- ROCm Libraries
   - [ ] rocALUTION rocalution
   - [ ] rocBLAS: rocblas
   - [ ] hipBLAS: hipblas
   - [ ] hipCUB: hipCUB
   - [ ] rocFFT: rocfft
   - [ ] rocRAND: rocrand
   - [ ] rocSPARSE: rocsparse
   - [ ] hipSPARSE: hipsparse
   - [ ] ROCm SMI Lib: rocm_smi_lib64
   - [ ] rocThrust: rocThrust
   - [ ] MIOpen: MIOpen-HIP (for the HIP version), MIOpen-OpenCL (for the OpenCL version)
   - [ ] MIOpenGEMM: miopengemm
   - [ ] RCCL: rccl
   - [ ] MIVisionX: mivisionx

**Comment by leofang** [comment]
Hi @isuruf Just wanna follow up here. What's the status of CF's support for the ROCm stack? I've been doing some PRs for CuPy and now the expected dependency will expand to almost all packages that you listed except for those with the prefix `MI`.

**Comment by isuruf** [comment]
They haven't been updated in a year now. I've dropped support for rocm in the latest builds because the patches didn't apply. You are welcome to help with updating llvm with the rocm patches.

**Comment by leofang** [comment]
Haha it's funny that I got similar suggestions from the Numba team (https://github.com/numba/numba/issues/6196). Could you give me a few pointers for how to proceed/what's needed so that I can play around at my spare time?

**Comment by isuruf** [comment]
:smile: I guess you already got the suggestion to look at  https://github.com/RadeonOpenCompute/llvm-project/

**Comment by leofang** [comment]
Yes I did! I mean the feedstocks. Which should I touch, llvmdev or llvm-meta (or both)? Do I apply the ROCm patches when building LLVM itself? I am unfamiliar with the llvm packages so need to know more.

**Comment by isuruf** [comment]
You can ignore `llvm-meta`. I think you need `llvmdev-feedstock` and `clangdev-feedstock`

**Comment by leofang** [comment]
Hi @isuruf I just took a look at `llvmdev-feedstock`, and I have a few follow-up questions: 
1. How exactly did you do to obtain the big patch https://github.com/conda-forge/llvmdev-feedstock/blob/master/recipe/patches/amd-roc-2.7.0.diff?
2. I noticed that we stopped applying the ROCm patch since https://github.com/conda-forge/llvmdev-feedstock/pull/85, but why was it? I couldn't find any discussion there.
3. I think for recent ROCm to work we need LLVM 11.0. Is there any plan? I saw there's a pending PR of yours (https://github.com/conda-forge/llvmdev-feedstock/pull/96) that could bring us closer.

**Comment by isuruf** [comment]
1. ROCm's LLVM repository moves with upstream LLVM git repository. So for ROCm 2.7.0 what I did was to figure out which commit they merged from upstream for 2.7.0 (this may not be a release) and then got the diff with that upstream commit. This diff was applied on the release branch. (For clang, what I used was https://github.com/conda-forge/clangdev-feedstock/blob/master/recipe/patches/amd-roc-hcc-2.7.0.diff#L1-L7)

2. No time

3. LLVM 11.0.0.rc3 will be released soon. Only 3 blockers right now.

**Comment by DanielWicz** [comment]
Is conda roc dead ?
